src/Matterhorn/State/Messages.hs

```diff
@@ -681,6 +686,57 @@ runNotifyCommand post mentioned = do
                                  [notified, sender, messageString] Nothing Nothing
                 return Nothing
 
+encodeToJSONstring :: A.ToJSON a => a -> String
+encodeToJSONstring a = map (chr . fromEnum) . BL.unpack $ A.encode a
+
+-- We define a notifyGetPayload for each notification version.
+notifyGetPayload :: Int -> ChatState -> Post -> Bool -> MH (Maybe String)
```

Comment:
Style note: this type signature should go on line 710 (immediately preceding the definition).

Reply:
Another thing I noticed about this function is that it is used in the `MH` monad, but that's unnecessary at present since all it does is `return`. It could be rewritten as a pure function (having type `... -> Maybe String`) and then used in a `let` binding below rather than with a monadic bind (`<-`).

Reply:
Resolved by commit 2b0420c0 .

Reply:
Resolved by commit f96534a .